.github/workflows/codeql.yml

```diff
@@ -0,0 +1,148 @@
+
+      node-version:
+        description: >
+          Node.js version to use for JavaScript/TypeScript analysis.
+          Ignored for other languages.
+        required: false
+        default: "22"
+        type: string
+
+      python-version:
+        description: >
+          Python version to use for Python analysis.
+          Ignored for other languages.
+        required: false
+        default: "3.11"
+        type: string
+
+      fail-fast:
+        description: >
+          Whether to stop all matrix jobs if one fails.
+        required: false
+        default: true
+        type: boolean
+
+      codeql-queries:
+        description: >
+          Additional CodeQL queries to run (comma-separated).
+          Example: "+security-and-quality,security-extended"
+        required: false
+        default: "+security-and-quality,security-extended"
+        type: string
+
+permissions:
+  contents: read
+  security-events: write
+  actions: read
+
+jobs:
+  codeql-analysis:
+    name: CodeQL
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: ${{ inputs.fail-fast }}
+      matrix:
+        language: ${{ fromJson(inputs.languages) }}
+
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}-${{ matrix.language }}
+      cancel-in-progress: true
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        if: matrix.language == 'javascript' || matrix.language == 'typescript'
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.node-version }}
+
+      - name: Cache Node.js dependencies
+        if: matrix.language == 'javascript' || matrix.language == 'typescript'
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.npm
+            node_modules
+          key: ${{ runner.os }}-node-${{ inputs.node-version }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ inputs.node-version }}-          
+
+      - name: Install Node.js dependencies (npm or yarn)
+        if: matrix.language == 'javascript' || matrix.language == 'typescript'
+        run: |
+          if [ -f package-lock.json ]; then
+            echo "Using npm ci"
+            npm ci
+          else
+            echo "No package-lock.json found. Using Yarn via Corepack..."
+            corepack enable
+            corepack prepare yarn@stable --activate
+            yarn install --immutable
+          fi
+
+      - name: Set up Python
+        if: matrix.language == 'python'
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python-version }}
+
+      - name: Cache Python dependencies
+        if: matrix.language == 'python'
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-python-${{ inputs.python-version }}-${{ hashFiles('**/python_requirements.txt') }}

```

Comment:
where is python_requirements.txt coming from? Usually this file is called requirements.txt, no?